promql/engine.go

```diff
@@ -1968,7 +1969,18 @@ var (
 )
 
 func getFPointSlice(sz int) []FPoint {
-	if p := fPointPool.Get(); p != nil {
+	index := int(math.Ceil(math.Log2(float64(sz)))) - minPoolSizePower
+
+	if index >= len(fPointPool) {
+		return make([]FPoint, 0, sz)
+	}
+
+	// if the size is < than the minPoolSizePower we return an array from the first pool
+	if index < 0 {
+		index = 0
+	}
+
+	if p := fPointPool[index].Get(); p != nil {
```

Comment:
How about we iterate from `index -> len(fPointPool)` and assign any available slice from a higher index instead of creating a new one?